node/queue.go

```diff
@@ -0,0 +1,29 @@
+package node
+
+import (
+	"github.com/virtual-kubelet/virtual-kubelet/internal/queue"
+)
+
+// These are exportable definitions of the queue package:
+
+// ShouldRetryFunc is a mechanism to have a custom retry policy
+//
+// it is passed metadata about the work item when the handler returns an error. It returns the following:
+// * The key
+// * The number of attempts that this item has already had (and failed)
+// * The (potentially wrapped) error from the queue handler.
+//
+// The return value is an error, and optionally an amount to delay the work.
+// If an error is returned, the work will be aborted, and the returned error is bubbled up. It can be the error that
+// was passed in or that error can be wrapped.
+//
+// If the work item should be is to be retried, a delay duration may be specified. If the value is nil, it will fall
+// back to the default behaviour of the queue, and use the ratelimiter that's configured to determine when to start work.
+type ShouldRetryFunc = queue.ShouldRetryFunc
```

Comment:
What happens when the returned delay is a negative value?

Reply:
Fixed.

Reply:
Can you point me to what changed? I'm not noticing.

Reply:
huh, no idea why it didn't update inline:

> 
> ShouldRetryFunc is a mechanism to have a custom retry policy
> 
>  it is passed metadata about the work item when the handler returns an error. It returns the following:
>  * The key
>  * The number of attempts that this item has already had (and failed)
>  * The (potentially wrapped) error from the queue handler.
> 
>  The return value is an error, and optionally an amount to delay the work.
>  If an error is returned, the work will be aborted, and the returned error is bubbled up. It can be the error that
>  was passed in or that error can be wrapped.
> 
>  If the work item should be is to be retried, a delay duration may be specified. The delay is used to schedule when
>  the item should begin processing relative to now, it does not necessarily dictate when the item will start work.
>  Items are processed in the order they are scheduled. If the delay is nil, it will fall  back to the default behaviour
>  of the queue, and use the rate limiter that's configured to determine when to start work.

Reply:
I'm not seeing where we are calling out what a negative value means, only nil.

Reply:
It doesn't call out the negative value explicitly, but it says that the ordering is based on now + delay. If the delay is negative, your task will be scheduled in the past, thus pushing it to the front of the queue. Do you think we need to be explicit about it?

Reply:
I think it is worth mentioning.

Reply:
Added: 
> // If the delay is negative, the item will be scheduled "earlier" than now. This will result in the item being executed
> // earlier than other items in the FIFO work order.